rules/TYPO312/v0/IgnorePageTypeRestrictionRector.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\TYPO312\v0;
+
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\ArrayItem;
+use PhpParser\Node\Scalar\String_;
+use Rector\Contract\Rector\ConfigurableRectorInterface;
+use Ssch\TYPO3Rector\Rector\AbstractTcaRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Webmozart\Assert\Assert;
+
+/**
+ * @changelog https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Deprecation-98487-ExtensionManagementUtilityallowTableOnStandardPages.html
+ */
+final class IgnorePageTypeRestrictionRector extends AbstractTcaRector implements ConfigurableRectorInterface
+{
+    public const TABLE_CONFIGURATION = 'table-configuration';
+
+    private ?string $table = null;
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Move method ExtensionManagementUtility::allowTableOnStandardPages to TCA configuration',
+            [new ConfiguredCodeSample(<<<'CODE_SAMPLE'
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+ExtensionManagementUtility::allowTableOnStandardPages('my_table');
+CODE_SAMPLE
+                , <<<'CODE_SAMPLE'
+$GLOBALS['TCA']['my_table']['ctrl']['security']['ignorePageTypeRestriction']
+CODE_SAMPLE
+                , [
+                    self::TABLE_CONFIGURATION => 'my_table',
```

Comment:
can you explain why it is useful to configure this and maybe even necessary? why can only one table be defined and not multiple?

Reply:
I have removed IgnorePageTypeRestrictionRector. It is not in use anymore. It was before in my draft. Good catch